RecoLocalTracker/SiPixelRecHits/interface/SiPixelTemplate2D.h

```diff
@@ -134,35 +174,62 @@ class SiPixelTemplate2D {
+   float scalex(int i) {
 #ifndef SI_PIXEL_TEMPLATE_STANDALONE
-      if(j < 0 || j > BXM3 || i < 0 || i > BYM3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::xytemp called with illegal indices = " << j << "," << i << std::endl;}
+      if(i < 0 || i > 3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::scalex called with illegal index = " << i << std::endl;}
 #else
-      assert((j>=0 && j<BYM2) && (i>=0 && i<BYM2));
+      assert(i>=0 && i<4);
 #endif
-      return xytemp_[j][i];} //!< current 2-d template
-   float chi2avg(int i) {
+      return scalex_[i];} //!< x-error scale factor in 4 charge bins
+   float scaley(int i) {
 #ifndef SI_PIXEL_TEMPLATE_STANDALONE
-      if(i < 0 || i > 3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::chi2yavg called with illegal index = " << i << std::endl;}
+      if(i < 0 || i > 3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::scaley called with illegal index = " << i << std::endl;}
 #else
       assert(i>=0 && i<4);
 #endif
-      return chi2avg_[i];} //!< average chi^2 in 4 charge bins
-   float chi2min(int i) {
+      return scaley_[i];} //!<x-error scale factor in 4 charge bins
+   float offsetx(int i) {
 #ifndef SI_PIXEL_TEMPLATE_STANDALONE
-      if(i < 0 || i > 3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::chi2ymin called with illegal index = " << i << std::endl;}
+      if(i < 0 || i > 3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::scaley called with illegal index = " << i << std::endl;}
 #else
       assert(i>=0 && i<4);
 #endif
-      return chi2min_[i];} //!< minimum chi^2 in 4 charge bins
-   float chi2avgone() {return chi2avgone_;}                        //!< //!< average y chi^2 for 1 pixel clusters
-   float chi2minone() {return chi2minone_;}                        //!< //!< minimum of y chi^2 for 1 pixel clusters
-   float lorywidth() {return lorywidth_;}                            //!< signed lorentz y-width (microns)
-   float lorxwidth() {return lorxwidth_;}                            //!< signed lorentz x-width (microns)
+      return offsetx_[i];} //!< x-offset in 4 charge bins
+   float offsety(int i) {
+#ifndef SI_PIXEL_TEMPLATE_STANDALONE
+      if(i < 0 || i > 3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::scaley called with illegal index = " << i << std::endl;}
+#else
+      assert(i>=0 && i<4);
+#endif
+      return offsety_[i];} //!< x-offset in 4 charge bins
+   float fbin(int i) {
+#ifndef SI_PIXEL_TEMPLATE_STANDALONE
+      if(i < 0 || i > 2) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::fbin called with illegal index = " << i << std::endl;}
+#else
+      assert(i>=0 && i<3);
+#endif
+      return fbin_[i];} //!< Return lower bound of Qbin definition
+   float sizex() {return clslenx_;}                             //!< return x size of template cluster
+   float sizey() {return clsleny_;}                             //!< return y size of template cluster
```

Comment:
this and many other methods that are not expected to change the object better be all declared as const ```float sizex() const ```

Reply:
Will discuss with Morris.  This may cascade elsewhere so he needs to be on board.  But the point is taken.